### PR TITLE
METRON-103: Unit Tests for storm bolts

### PR DESCRIPTION
**Pull request by merrimanr**
Unit tests for all bolts used in the parser and enrichment topologies.  Tested with Integration tests, unit tests and single-node vagrant.

### CODE REVIEW

File: deployment/roles/metron_ui/tasks/main.yml [hunk 1]
```diff
@@ -53,4 +53,4 @@
     production: no
 
 - name: Start Metron UI
-  shell: "pm2 restart {{ metron_ui_directory }}/lib/metron-ui.js --name metron"
+  shell: "pm2 start {{ metron_ui_directory }}/lib/metron-ui.js --name metron"
```
Comment (dlyle65535) [inline deployment/roles/metron_ui/tasks/main.yml]: Did you intend to revert this change?

Comment (merrimanr) [inline deployment/roles/metron_ui/tasks/main.yml]: Yes this was intentional.  Turns out the previous commit only works on AWS so I reverted it to keep the single node vagrant deployment from breaking.  @nickwallen reopened the pull request and will submit another fix.  Sorry I should have called that out in the description.

Comment (dlyle65535) [inline deployment/roles/metron_ui/tasks/main.yml]: Oh, good call then. :)

**Comment by dlyle65535** [comment]
+1 Great stuff, ship it! :)

File: metron-streaming/Metron-Common/src/test/java/org/apache/metron/bolt/BulkMessageWriterTest.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    }
+    when(tuple.getValueByField("message")).thenReturn(messageList.get(4));
+    tupleList.add(tuple);
+    bulkMessageWriterBolt.execute(tuple);
+    verify(bulkMessageWriter, times(1)).write(eq(sensorType), any(Configurations.class), eq(tupleList), argThat(new MessageListMatcher(messageList)));
+    verify(outputCollector, times(5)).ack(tuple);
+    reset(outputCollector);
+    doThrow(new Exception()).when(bulkMessageWriter).write(eq(sensorType), any(Configurations.class), Matchers.anyListOf(Tuple.class), Matchers.anyListOf(JSONObject.class));
+    when(tuple.getValueByField("message")).thenReturn(messageList.get(0));
+    for(int i = 0; i < 5; i++) bulkMessageWriterBolt.execute(tuple);
```
Comment (cestella) [inline metron-streaming/Metron-Common/src/test/java/org/apache/metron/bolt/BulkMessageWriterTest.java]: This is a style thing, but can we please have braces?  It's so easy to accidentally screw if's and for's up without explicit braces.

Comment (merrimanr) [inline metron-streaming/Metron-Common/src/test/java/org/apache/metron/bolt/BulkMessageWriterTest.java]: Fixed.  Also realized I named the test wrong so I fixed that as well.

File: metron-streaming/Metron-EnrichmentAdapters/src/test/java/org/apache/metron/enrichment/bolt/EnrichmentJoinBoltTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class EnrichmentJoinBoltTest extends BaseEnrichmentBoltTest {
+
+  /**
+   * {
```
Comment (cestella) [inline metron-streaming/Metron-EnrichmentAdapters/src/test/java/org/apache/metron/enrichment/bolt/EnrichmentJoinBoltTest.java]: I think those *'s are going to show up in your string, is that what you intended?

Comment (cestella) [inline metron-streaming/Metron-EnrichmentAdapters/src/test/java/org/apache/metron/enrichment/bolt/EnrichmentJoinBoltTest.java]: Nevermind, they don't.  retracted

**Comment by cestella** [comment]
+1, looks good
